core/providers/azure/images.go

```diff
@@ -0,0 +1,302 @@
+
+	providerName := provider.GetProviderKey()
+
+	jsonData, err := sonic.Marshal(azureRequest)
+	if err != nil {
+		return nil, "", 0, providerUtils.NewBifrostOperationError("could not serialize azure image generation request", err, providerName)
+	}
+
+	response, deployment, latency, bifrostErr := provider.completeRequest(ctx, jsonData, "images/generations", key, azureRequest.Model, schemas.ImageGenerationRequest)
+
+	// Handle error response
+	if bifrostErr != nil {
+		return nil, "", 0, bifrostErr
+	}
+	azureResponse := &AzureImageResponse{}
+	if err := sonic.Unmarshal(response, azureResponse); err != nil {
+		return nil, "", 0, providerUtils.NewBifrostOperationError("failed to unmarshal Azure image response", err, schemas.Azure)
+	}
+
+	return azureResponse, deployment, latency, nil
+
+}
+
+// ImageGenerationStream performs a streaming image generation request to Azure's API.
+// It formats the request, sends it to Azure, and processes the response.
+// Returns a channel of BifrostStream objects or an error if the request fails.
+func (provider *AzureProvider) ImageGenerationStream(
+	ctx context.Context,
+	postHookRunner schemas.PostHookRunner,
+	key schemas.Key,
+	request *schemas.BifrostImageGenerationRequest,
+) (chan *schemas.BifrostStream, *schemas.BifrostError) {
+
+	// Validate api key configs
+	if err := provider.validateKeyConfig(key); err != nil {
+		return nil, err
+	}
+
+	//
+	deployment := key.AzureKeyConfig.Deployments[request.Model]
+	if deployment == "" {
+		return nil, providerUtils.NewConfigurationError(fmt.Sprintf("deployment not found for model %s", request.Model), provider.GetProviderKey())
+	}
+
+	apiVersion := key.AzureKeyConfig.APIVersion
+	if apiVersion == nil {
+		apiVersion = schemas.Ptr(AzureAPIVersionDefault)
+	}
+
+	url := fmt.Sprintf("%s/openai/deployments/%s/images/generations?api-version=%s", key.AzureKeyConfig.Endpoint, deployment, *apiVersion)
+
+	// Prepare Azure-specific headers
+	authHeader := make(map[string]string)
+
+	// Set Azure authentication - either Bearer token or api-key
+	if authToken, ok := ctx.Value(AzureAuthorizationTokenKey).(string); ok {
+		authHeader["Authorization"] = fmt.Sprintf("Bearer %s", authToken)
+	} else {
+		authHeader["api-key"] = key.Value
+	}
+
+	if !openai.StreamingEnabledImageModels[request.Model] {
+		return provider.simulateImageStreaming(ctx, postHookRunner, key, request)
+	}
+
+	// Azure is OpenAI-compatible
+	return openai.HandleOpenAIImageGenerationStreaming(
+		ctx,
+		provider.client,
+		url,
+		request,
+		authHeader,
+		provider.networkConfig.ExtraHeaders,
+		providerUtils.ShouldSendBackRawResponse(ctx, provider.sendBackRawResponse),
+		provider.GetProviderKey(),
+		postHookRunner,
+		nil,
+		nil, nil,
+		provider.logger,
+	)
+
+}
+
+func (provider *AzureProvider) simulateImageStreaming(
```

Comment:
you can directly throw an error for streaming call if incoming models dont support streaming (like we do for speech/transcription methods) no need to simulate streaming